src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -1409,7 +1412,7 @@ private static void validateViolationSection(String fileName, String sectionName
                 assertWithMessage(fileName + " section '" + sectionName
                         + "' should have matching url for '" + linkText + "'")
                     .that(url)
-                    .isEqualTo(expectedUrl);
+                    .contains(expectedUrl);
```

Comment:
Why is this not equal to anymore? The point of `equalTo` was to ensure we have the correct URL entered, with no mistakes.

Reply:
Here is a failure in `testAllCheckSections` with `isEqualTo(expectedUrl)`
```
methodparampad.xml section 'MethodParamPad' should have matching url for 'see the documentation'
expected: config.html#Custom_messages
but was : ../../config.html#Custom_messages
Expected :config.html#Custom_messages
Actual   :../../config.html#Custom_messages
```

On second thought, I should have done this and not touch the `isEqualsTo()`
```diff
-                    expectedUrl = "config.html#Custom_messages";
+                    expectedUrl = "../../config.html#Custom_messages";
```

Done.

Reply:
Its fine until it becomes an issue if you don't account for the folder depth your in.